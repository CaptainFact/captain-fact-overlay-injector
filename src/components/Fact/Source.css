.sourceLink {
  text-decoration-color: lightgrey;
  display: block;
  text-shadow: 0 0 1px rgba(0, 0, 0, 0.15);
  text-decoration: none;
  padding: 0.5em;
}

.sourceLink img {
  height: 1em;
}

.sourceLink:hover .articleTitle {
  text-decoration: underline;
}

.siteName {
  text-decoration: underline;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: white;
  font-weight: bold;
  font-size: 0.9em;
  vertical-align: top;
}

.articleTitle {
  font-size: 0.9em;
  color: #e0e0e0;
  text-decoration-color: lightgrey;
}
