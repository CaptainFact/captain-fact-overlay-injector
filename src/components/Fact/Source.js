import React from 'react'
import upperCase from 'voca/upper_case'

import { siteNameContainer, siteName, articleTitle, sourceLink } from './Source.css'
import DEFAULT_IMG_NEW_TAB from '../../assets/new_tab.png'


const getHostName = url =>
  upperCase(url.replace(/https?:\/\//, '').replace(/\/.*/g, ''))

const Source = ({source: {url, title, site_name}, imgNewTab = DEFAULT_IMG_NEW_TAB}) => (
  <a href={url} target="_BLANK" className={sourceLink}>
    <div className={siteNameContainer}>
      <img src={imgNewTab}/>&nbsp;
      <span className={siteName}>
        { upperCase(site_name) || getHostName(url) }
      </span>
    </div>
    <span className={articleTitle}>{title}</span>
  </a>
)

export default Source
