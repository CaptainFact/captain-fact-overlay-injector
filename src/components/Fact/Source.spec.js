import React from 'react'
import Source from './Source'

const DEFAULT_SOURCE = {
  url: 'https://truetruth.local/this-is-the-truth',
  title: 'The true fact about facts',
  site_name: 'TrueTruth',
}

test('full source', () => {
  snapshot(<Source source={DEFAULT_SOURCE} />)
})

test('source without site name', () => {
  snapshot(<Source source={{ ...DEFAULT_SOURCE, site_name: null }} />)
})

test('with custom new tab image', () => {
  snapshot(<Source source={DEFAULT_SOURCE} imgNewTab="custom-pic.jpg" />)
})
