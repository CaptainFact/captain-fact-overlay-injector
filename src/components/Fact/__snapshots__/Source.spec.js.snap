// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full source 1`] = `
<a
  className="sourceLink"
  href="https://truetruth.local/this-is-the-truth"
  target="_BLANK"
>
  <div>
    <img
      src="test-file-stub"
    />
     
    <span
      className="siteName"
    >
      TRUETRUTH
    </span>
  </div>
  <span
    className="articleTitle"
  >
    The true fact about facts
  </span>
</a>
`;

exports[`source without site name 1`] = `
<a
  className="sourceLink"
  href="https://truetruth.local/this-is-the-truth"
  target="_BLANK"
>
  <div>
    <img
      src="test-file-stub"
    />
     
    <span
      className="siteName"
    >
      TRUETRUTH.LOCAL
    </span>
  </div>
  <span
    className="articleTitle"
  >
    The true fact about facts
  </span>
</a>
`;

exports[`with custom new tab image 1`] = `
<a
  className="sourceLink"
  href="https://truetruth.local/this-is-the-truth"
  target="_BLANK"
>
  <div>
    <img
      src="custom-pic.jpg"
    />
     
    <span
      className="siteName"
    >
      TRUETRUTH
    </span>
  </div>
  <span
    className="articleTitle"
  >
    The true fact about facts
  </span>
</a>
`;
